lib/jekyll/theme.rb

```diff
@@ -62,11 +62,22 @@ def realpath_for(folder)
       # escape the theme root.
       # However, symlinks are allowed to point to other directories within the theme.
       Jekyll.sanitized_path(root, File.realpath(Jekyll.sanitized_path(root, folder.to_s)))
-    rescue Errno::ENOENT, Errno::EACCES, Errno::ELOOP
-      Jekyll.logger.warn "Invalid theme folder:", folder
+    rescue Errno::ENOENT, Errno::EACCES, Errno::ELOOP => e
```

Comment:
I think it makes more sense here to not rescue all three exceptions, but to rescue them individually, so that it's more explicit what's happening here without the extra indirection of the method call
```suggestion
    rescue Errno::ENOENT
      # it's ok for things to not exist inside of the theme
    end
    rescue Errno::EACCES => e
      Jekyll.logger.error "Theme error:", "Directory '#{folder}' is not accessible."
    end
    rescue Errno::ELOOP => e
      Jekyll.logger.error "Theme error:", "Directory '#{folder}' includes a symbolic link loop."
    end
```

We can do that now, save it for a later PR, or just let it drop. I'm not going to be that picky about it.

Reply:
👍 👍 

Then every rescue should include a `nil` so it doesn't return a truthy, right? (see https://github.com/jekyll/jekyll/pull/5195/commits/0278abe3f95a74e07ddd5334b6de8f167aaa5056)

Reply:
There's a chance that RuboCop might complain about having too many lines in a method definition or perhaps some other `Metric` cop flags in.. So let us defer this change for a future PR.